onchain-program/src/processor.rs

```diff
@@ -137,9 +132,23 @@ impl Processor {
         msg!("creating the data bucket");
 
         invoke_signed(
-            &system_instruction::create_account(&payer_key, &data_bucket_key, required_lamports, data_bucket_len as u64, program_id),
-            &[data_bucket_account.clone()],
-            &[&[b"solana-data-packer", authority_key.as_ref()], &[&[bump_seed]]],
+            &system_instruction::create_account(
+                &payer_key,
+                &data_bucket_key,
+                required_lamports,
+                data_bucket_len as u64,
+                program_id,
+            ),
+            &[
+                data_bucket_account.clone(),
+                // the signer of this forwarded invoke instruction
+                // just so happens to be the same account as the authority_key
+                payer_account.clone(),
+                // we need to account of the program being called
+                system_program_account.clone(),
```

Comment:
Oh! Good catch on this one!

Looking at https://docs.rs/solana-program/1.9.8/solana_program/system_instruction/enum.SystemInstruction.html#variant.CreateAccount

Should the `data_bucket_account` and `payer_account` actually be other way round?

Why is the `system_program_account` needed, but not specified in the documentation?

Reply:
> Should the data_bucket_account and payer_account actually be other way round?
you mean the order? I don't think that matters. 

> system_program_account
I think this is to be sure that the user signs it and knows what is being used as the permission is "forwarded".